docs/developer-docs/smart-contracts/call-overview.mdx

```diff
@@ -0,0 +1,161 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Overview 
+
+To interact with a canister smart contract's methods, there are two primary types of calls that can be used: update calls and query calls.
+
+Canisters implement methods that can be called via update or query calls. A query method can be called as both an update and a query, whereas update methods can be called only as an update.
+
+**Update calls** can make modifications to a canister's state, while **query calls** cannot.
+
+<div class="text--center">
+<img src="/img/docs/update_vs_query.png" alt="Update vs Query" width="400"/>
+</div>
+
+## Update calls
+
+Update calls are executed on all nodes of a subnet since the result must go through consensus. This makes them slower compared to query calls. They are submitted and answered asynchronously.
+
+### Example update call
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+actor countCharacters {
+    public func test(text : Text) : async Bool {
+        let size = Text.size(text);
+        return size % 2 == 0;
+    };
+};
+```
+
+</TabItem>
+
+<TabItem value="rust" label="Rust">
+
+```rust
+#[ic_cdk_macros::update]
+fn increment() {
+    COUNTER.with(|counter| *counter.borrow_mut() += 1);
+}
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```typescript
+setMessage: update([text], Void, (newMessage) => {
+    message = newMessage; // This change will be persisted
+})
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+@update
+def set_message(new_message: str) -> void:
+    global message
+    message = new_message
+```
+
+</TabItem>
+</Tabs>
+
+## Query calls
+
+Query calls are executed on a single node and return a synchronous response. Since they execute on a single node, they do not go through consensus and can be much faster than update calls.
```

Comment:
I'd recommend a warning here that a single malicious node could lie about a response, and an update call or a certified query (link) should be used for security-critical calls.

Reply:
Good point, I added a warning.